src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -643,17 +643,21 @@ private CheckSumAndSizeWriteStream GetDataCompressor(Stream backingStream, Boole
             Boolean isIntermediateStream = true;
 
             Boolean leaveCompressorStreamOpenOnClose = leaveBackingStreamOpen && !isIntermediateStream;
-            CheckSumAndSizeWriteStream checkSumStream =
-                new CheckSumAndSizeWriteStream(compressorStream, backingStream, leaveCompressorStreamOpenOnClose,
-                    (Int64 initialPosition, Int64 currentPosition, UInt32 checkSum) =>
-                    {
-                        _crc32 = checkSum;
-                        _uncompressedSize = currentPosition;
-                        _compressedSize = backingStream.Position - initialPosition;
+            var checkSumStream = new CheckSumAndSizeWriteStream(
+                compressorStream, 
+                backingStream, 
+                leaveCompressorStreamOpenOnClose, 
+                this, 
+                onClose,
+                (Int64 initialPosition, Int64 currentPosition, UInt32 checkSum, Stream backing, ZipArchiveEntry entry, EventHandler onclose) =>
```

Comment:
Same comment as on the other PR: maybe ```thisRef``` or ```innerThis``` or something like that instead of ```entry```.  I think it just better highlights that we want to be using ```this```, but can't due to the resulting closure.

And also same comment about not shadowing the ```onClose``` name from the outer scope.